raiden/network/transport/matrix/rtc/web_rtc.py

```diff
@@ -351,7 +348,9 @@ async def _close(self) -> None:
         self._ice_connection_closed(self)
 
     def close(self) -> Task:
-        return asyncio.create_task(self._close())
+        if self._closing_task is None:
+            self._closing_task = asyncio.create_task(self._close())
+        return self._closing_task
```

Comment:
Why do we keep a reference to the closing task? We don't reset it anywhere from what I can tell.
Would it not be enough to just have a boolean, like before, but here?

Reply:
Because close is called from different points in the code. If it was called somewhere and then during the close another close is called and waited on, then it would not actually be waited on the task which closes the connection instead it would wait on the task which directly stops because closing would be true. 
The code would continue but the actual close would not be finished yet